library/src/main/java/com/datatorrent/lib/fileaccess/FileAccess.java

```diff
@@ -151,7 +151,7 @@
      * @param value
      * @throws IOException
      */
-    void append(byte[] key, byte[] value) throws IOException;
+    void append(Slice key, Slice value) throws IOException;
```

Comment:
@brightchen I understand that this interface is evolving so making backward incompatible change with next version is okay. However, this interface exists for a long time. Is it better here to deprecate old methods and add new one?
In some cases, we have broken backward compatibility but the reason there was that all the api methods changed, not just one or two, and the change was quite drastic that existing implementation couldn't be re-used. Deprecating old ones would have just made the code messier.
On the other hand, if this is just being used by utilities in Malhar, we can break backward compatibility.  I think it will be good to get @tweise  input on this.